cpp/scenario/gazebo/CMakeLists.txt

```diff
@@ -0,0 +1,119 @@
+# Copyright (C) 2020 Istituto Italiano di Tecnologia (IIT)
+# All rights reserved.
+#
+#  This project is dual licensed under LGPL v2.1+ or Apache License.
+#
+# -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -
+#
+#  This software may be modified and distributed under the terms of the
+#  GNU Lesser General Public License v2.1 or any later version.
+#
+# -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+# ================
+# Extra Components
+# ================
+
+set(EXTRA_COMPONENTS_HDRS
+#    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/ECMMutex.h
+#    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/PreUpdateConditionVariable.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/JointPID.h
+#    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/Constraint.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/SimulatedTime.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/BasePoseTarget.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/BaseWorldVelocityTarget.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/BaseWorldAccelerationTarget.h
+#    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/InitialWorldSdf.h
+#    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/InitialModelSdf.h
+#    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/ConstraintsData.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/JointEffortLimit.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/JointControlMode.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/WorldVelocityCmd.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/JointPositionReset.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/JointVelocityReset.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/JointPositionTarget.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/JointVelocityTarget.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/JointAccelerationTarget.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/scenario/gazebo/components/JointControllerPeriod.h)
```

Comment:
Just to clarify, this components are unused at the moment, right?

Reply:
I am trying to split the refactoring in chunks. These components will be read by joint controller plugins that I will propose in a later PR (I'm trying to propose new features sequentially).